src/components/data/WdsTroubleshooter.js

```diff
@@ -0,0 +1,202 @@
+    'Data Table app running?':	appRunning,
+    'Data Table app proxy url':	proxyUrl,
+    'Data Table app responding':	wdsResponsive,
+    'Data Table app version':	version,
+    'Data Table app status':	wdsStatus,
+    'Data Table app DB status':	wdsDbStatus,
+    'Data Table app ping status':	wdsPingStatus,
+    'Default Instance exists': defaultInstanceExists
+  }
+
+  return h(Modal, {
+    showCancel: false,
+    onDismiss,
+    title: 'Data Table Troubleshooter',
+    width: '55rem',
+    okButton: h(ButtonPrimary, {
+      tooltip: 'Done',
+      onClick: onDismiss
+    }, ['Done'])
+  }, [div({ style: { padding: '1rem 0.5rem', lineHeight: '1.4rem' } }, [
+    table({ style: { borderSpacing: '1rem 0', borderCollapse: 'separate' } }, [
+      tr([
+        td({ style: { fontWeight: 'bold' } }, [
+          !!workspaceId ? checkIcon('success') : checkIcon('failure')
+        ]),
+        td({ style: { fontWeight: 'bold' } }, ['Workspace Id']),
+        td([clippy('Workspace Id', workspaceId)])
+      ]),
+      tr([
+        td({ style: { fontWeight: 'bold' } }, [
+          !!mrgId ? checkIcon('success') : checkIcon('failure')
+        ]),
+        td({ style: { fontWeight: 'bold' } }, ['Resource Group Id']),
+        td([clippy('Resource Group Id', mrgId)])
+      ]),
+      tr([
+        td({ style: { fontWeight: 'bold' } }, [leoOk == null ? checkIcon('running') :
+          (!!leoOk?.length ? checkIcon('success') : checkIcon('failure'))]),
+        td({ style: { fontWeight: 'bold' } }, ['App listing']),
+        td([h(Fragment, [`${leoOk?.length} app(s) total`])])
+      ]),
+      tr([
+        td({ style: { fontWeight: 'bold' } }, [leoOk == null ? checkIcon('running') :
+          (!!appFound && appFound !== 'unknown' ? checkIcon('success') : checkIcon('failure'))]),
+        td({ style: { fontWeight: 'bold' } }, ['Data Table app found']),
+        td([clippy('found app name', appFound)])
+      ]),
+      tr([
+        td({ style: { fontWeight: 'bold' } }, [appRunning == null ? checkIcon('running') :
+          (!!appRunning && appRunning !== 'unknown' ? checkIcon('success') : checkIcon('failure'))]),
+        td({ style: { fontWeight: 'bold' } }, ['Data Table app status']),
+        td([h(Fragment, [appRunning])])
+      ]),
+      tr([
+        td({ style: { fontWeight: 'bold' } }, [proxyUrl == null ? checkIcon('running') :
+          (!!proxyUrl && proxyUrl !== 'unknown' ? checkIcon('success') : checkIcon('failure'))]),
+        td({ style: { fontWeight: 'bold' } }, ['Data Table app proxy url']),
+        // don't use the clippy() helper here so we can truncate the proxyUrl
+        td([
+          h(Fragment, [h(div, { style: _.merge({ width: '400px', float: 'left' }, Style.noWrapEllipsis) }, [proxyUrl]),
+            h(ClipboardButton, {
+              'aria-label': 'Copy proxy url to clipboard',
+              style: { marginLeft: '1rem' },
+              text: proxyUrl
+            })])
+        ])
+      ]),
+      tr([
```

Comment:
Minor nit, but I think it might be cleaner to refactor each of these rows into a simple child component that passes in the values needed for the logic as args.?

It seems to be a lot of repetitive rows / updating anything might be a bit of a pain.

Reply:
Good call.  Could also help with the repetition of the dictionary I made for text values.

Reply:
totally agree, I had the same thought while typing out those repetitive rows